File: Python3/implement_queue_using_stacks.py
class MyQueue:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.stack = []

    def push(self, x: int) -> None:
        """
        Push element x to the back of queue.
        """
        self.stack.append(x)

    def pop(self) -> int:
        """
        Removes the element from in front of queue and returns that element.
        """
        if len(self.stack) == 0: return -1
        
        front_num = self.stack[0]
        self.stack.remove(front_num)
        
        return front_num

    def peek(self) -> int:
        """
        Get the front element.
        """
        if len(self.stack) == 0: return -1
        
        return self.stack[0]

    def empty(self) -> bool:
        """
        Returns whether the queue is empty.
        """
        return not len(self.stack)


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()

File: C++/implement_queue_using_stacks.cpp
class MyQueue {
    vector<int> stack;
    
public:
    /** Initialize your data structure here. */
    MyQueue() {
        
    }
    
    /** Push element x to the back of queue. */
    void push(int x) {
        stack.push_back(x);
    }
    
    /** Removes the element from in front of queue and returns that element. */
    int pop() {
        if(stack.size() == 0) return -1;
        
        int front_num = stack[0];
        stack.erase (stack.begin());
        
        return front_num;
    }
    
    /** Get the front element. */
    int peek() {
        if(stack.size() == 0) return -1;
        
        return stack[0];
    }
    
    /** Returns whether the queue is empty. */
    bool empty() {
        return stack.size() == 0;
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */
